### PR TITLE
save array creations on each raycaster update

### PR DESCRIPTION
**Pull request by ngokevin**
**Description:**

`flattenChildrenShallow` was creating two arrays on each raycaster refresh. And raycasters auto refresh every time there is an attribute change or child append on the scene. And a scene has often multiple raycasters. So this saves quite a ton of arrays.

**Changes proposed:**
- Reuse arrays for raycaster flattenChildrenShallow.

### CODE REVIEW

File: src/components/raycaster.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ module.exports.Component = registerComponent('raycaster', {
     direction: {type: 'vec3', default: {x: 0, y: 0, z: -1}},
     enabled: {default: true},
     far: {default: 1000},
-    interval: {default: 0},
+    interval: {default: 100},
```
Comment (vincentfretin) [inline src/components/raycaster.js]: I don't think you meant to revert this change.

Comment (ngokevin) [reply src/components/raycaster.js]: I'll do in another commit, i think 100 was appropriate for perf
